### PR TITLE
health, ipn/ipnlocal: track, log overall health

### PR DESCRIPTION
**Pull request by bradfitz**
Updates #1505

### CODE REVIEW

File: health/health.go [hunk 1]
```diff
@@ -32,21 +35,35 @@ var (
 	ipnWantRunning          bool
 )
 
+// ErrorKey is an overall category for which an error is being reported.
+type ErrorKey string
+
+const (
+	KeyOverall = ErrorKey("overall")
```
Comment (crawshaw) [inline health/health.go:42]: I find this surprisingly confusing. Maybe there's some obvious way of thinking about this that makes it make sense?

Maybe it needs some docs?

Comment (bradfitz) [reply health/health.go:42]: Whoops, my docs on ErrorKey itself also used word "overall" by accident, without realizing it conflicted with this "overall", which is different. I'll reword there and add docs here.

Comment (bradfitz) [reply health/health.go]: Done. PTAL.

**Review by crawshaw — APPROVED** [review]
Makes sense now, thanks. Is this new timer going to fire regularly on iOS?

**Comment by bradfitz** [comment]
> Makes sense now, thanks. Is this new timer going to fire regularly on iOS?

Yes, but no more often than regular heartbeats from control+DERP, and it won't log anything if nothing's changed.

**Pull request merged**